imagenet/main.py

```diff
@@ -200,16 +232,29 @@ def train(train_loader, model, criterion, optimizer, epoch):
         top1.update(prec1[0], input.size(0))
         top5.update(prec5[0], input.size(0))
 
+        loss = loss*args.loss_scale
         # compute gradient and do SGD step
-        optimizer.zero_grad()
-        loss.backward()
-        optimizer.step()
+
+        if args.fp16:
+            model.zero_grad()
+            loss.backward()
+            if args.loss_scale != 1:
```

Comment:
You could just do `(loss / args.loss_scale).backward()` and it would be much faster. BTW it's weird that it's in fact _inverse scaling_ of the loss